File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/WeatherStation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class WeatherStation : ISerialized
    {
        public WeatherStation(XmlNode xmlNode)
        {
            XmlNode weatherStationCityNode = xmlNode.SelectSingleNode("weatherStationCity");
            
            if (weatherStationCityNode != null)
            {
                if (weatherStationCityNode.Attributes["href"] != null || weatherStationCityNode.Attributes["id"] != null) 
                {
                    if (weatherStationCityNode.Attributes["id"] != null) 
                    {
                        weatherStationCityIDRef_ = weatherStationCityNode.Attributes["id"].Value;
                        BusinessCenter ob = new BusinessCenter(weatherStationCityNode);
                        IDManager.SetID(weatherStationCityIDRef_, ob);
                    }
                    else if (weatherStationCityNode.Attributes["href"] != null)
                    {
                        weatherStationCityIDRef_ = weatherStationCityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        weatherStationCity_ = new BusinessCenter(weatherStationCityNode);
                    }
                }
                else
                {
                    weatherStationCity_ = new BusinessCenter(weatherStationCityNode);
                }
            }
            
        
            XmlNode weatherStationAirportNode = xmlNode.SelectSingleNode("weatherStationAirport");
            
            if (weatherStationAirportNode != null)
            {
                if (weatherStationAirportNode.Attributes["href"] != null || weatherStationAirportNode.Attributes["id"] != null) 
                {
                    if (weatherStationAirportNode.Attributes["id"] != null) 
                    {
                        weatherStationAirportIDRef_ = weatherStationAirportNode.Attributes["id"].Value;
                        WeatherStationAirport ob = new WeatherStationAirport(weatherStationAirportNode);
                        IDManager.SetID(weatherStationAirportIDRef_, ob);
                    }
                    else if (weatherStationAirportNode.Attributes["href"] != null)
                    {
                        weatherStationAirportIDRef_ = weatherStationAirportNode.Attributes["href"].Value;
                    }
                    else
                    {
                        weatherStationAirport_ = new WeatherStationAirport(weatherStationAirportNode);
                    }
                }
                else
                {
                    weatherStationAirport_ = new WeatherStationAirport(weatherStationAirportNode);
                }
            }
            
        
            XmlNode weatherStationWBANNode = xmlNode.SelectSingleNode("weatherStationWBAN");
            
            if (weatherStationWBANNode != null)
            {
                if (weatherStationWBANNode.Attributes["href"] != null || weatherStationWBANNode.Attributes["id"] != null) 
                {
                    if (weatherStationWBANNode.Attributes["id"] != null) 
                    {
                        weatherStationWBANIDRef_ = weatherStationWBANNode.Attributes["id"].Value;
                        WeatherStationWBAN ob = new WeatherStationWBAN(weatherStationWBANNode);
                        IDManager.SetID(weatherStationWBANIDRef_, ob);
                    }
                    else if (weatherStationWBANNode.Attributes["href"] != null)
                    {
                        weatherStationWBANIDRef_ = weatherStationWBANNode.Attributes["href"].Value;
                    }
                    else
                    {
                        weatherStationWBAN_ = new WeatherStationWBAN(weatherStationWBANNode);
                    }
                }
                else
                {
                    weatherStationWBAN_ = new WeatherStationWBAN(weatherStationWBANNode);
                }
            }
            
        
            XmlNode weatherStationWMONode = xmlNode.SelectSingleNode("weatherStationWMO");
            
            if (weatherStationWMONode != null)
            {
                if (weatherStationWMONode.Attributes["href"] != null || weatherStationWMONode.Attributes["id"] != null) 
                {
                    if (weatherStationWMONode.Attributes["id"] != null) 
                    {
                        weatherStationWMOIDRef_ = weatherStationWMONode.Attributes["id"].Value;
                        WeatherStationWMO ob = new WeatherStationWMO(weatherStationWMONode);
                        IDManager.SetID(weatherStationWMOIDRef_, ob);
                    }
                    else if (weatherStationWMONode.Attributes["href"] != null)
                    {
                        weatherStationWMOIDRef_ = weatherStationWMONode.Attributes["href"].Value;
                    }
                    else
                    {
                        weatherStationWMO_ = new WeatherStationWMO(weatherStationWMONode);
                    }
                }
                else
                {
                    weatherStationWMO_ = new WeatherStationWMO(weatherStationWMONode);
                }
            }
            
        
        }
        
    
        #region WeatherStationCity_
        private BusinessCenter weatherStationCity_;
        public BusinessCenter WeatherStationCity_
        {
            get
            {
                if (this.weatherStationCity_ != null)
                {
                    return this.weatherStationCity_; 
                }
                else if (this.weatherStationCityIDRef_ != null)
                {
                    weatherStationCity_ = IDManager.getID(weatherStationCityIDRef_) as BusinessCenter;
                    return this.weatherStationCity_; 
                }
                else
                {
                      return this.weatherStationCity_; 
                }
            }
            set
            {
                if (this.weatherStationCity_ != value)
                {
                    this.weatherStationCity_ = value;
                }
            }
        }
        #endregion
        
        public string weatherStationCityIDRef_ { get; set; }
        #region WeatherStationAirport_
        private WeatherStationAirport weatherStationAirport_;
        public WeatherStationAirport WeatherStationAirport_
        {
            get
            {
                if (this.weatherStationAirport_ != null)
                {
                    return this.weatherStationAirport_; 
                }
                else if (this.weatherStationAirportIDRef_ != null)
                {
                    weatherStationAirport_ = IDManager.getID(weatherStationAirportIDRef_) as WeatherStationAirport;
                    return this.weatherStationAirport_; 
                }
                else
                {
                      return this.weatherStationAirport_; 
                }
            }
            set
            {
                if (this.weatherStationAirport_ != value)
                {
                    this.weatherStationAirport_ = value;
                }
            }
        }
        #endregion
        
        public string weatherStationAirportIDRef_ { get; set; }
        #region WeatherStationWBAN_
        private WeatherStationWBAN weatherStationWBAN_;
        public WeatherStationWBAN WeatherStationWBAN_
        {
            get
            {
                if (this.weatherStationWBAN_ != null)
                {
                    return this.weatherStationWBAN_; 
                }
                else if (this.weatherStationWBANIDRef_ != null)
                {
                    weatherStationWBAN_ = IDManager.getID(weatherStationWBANIDRef_) as WeatherStationWBAN;
                    return this.weatherStationWBAN_; 
                }
                else
                {
                      return this.weatherStationWBAN_; 
                }
            }
            set
            {
                if (this.weatherStationWBAN_ != value)
                {
                    this.weatherStationWBAN_ = value;
                }
            }
        }
        #endregion
        
        public string weatherStationWBANIDRef_ { get; set; }
        #region WeatherStationWMO_
        private WeatherStationWMO weatherStationWMO_;
        public WeatherStationWMO WeatherStationWMO_
        {
            get
            {
                if (this.weatherStationWMO_ != null)
                {
                    return this.weatherStationWMO_; 
                }
                else if (this.weatherStationWMOIDRef_ != null)
                {
                    weatherStationWMO_ = IDManager.getID(weatherStationWMOIDRef_) as WeatherStationWMO;
                    return this.weatherStationWMO_; 
                }
                else
                {
                      return this.weatherStationWMO_; 
                }
            }
            set
            {
                if (this.weatherStationWMO_ != value)
                {
                    this.weatherStationWMO_ = value;
                }
            }
        }
        #endregion
        
        public string weatherStationWMOIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/WeatherStation.cpp
// WeatherStation.cpp 
#include "WeatherStation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

WeatherStation::WeatherStation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //weatherStationCityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* weatherStationCityNode = xmlNode->FirstChildElement("weatherStationCity");

   if(weatherStationCityNode){weatherStationCityIsNull_ = false;}
   else{weatherStationCityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- weatherStationCityNode , address : " << weatherStationCityNode << std::endl;
   #endif
   if(weatherStationCityNode)
   {
      if(weatherStationCityNode->Attribute("href") || weatherStationCityNode->Attribute("id"))
      {
          if(weatherStationCityNode->Attribute("id"))
          {
             weatherStationCityIDRef_ = weatherStationCityNode->Attribute("id");
             weatherStationCity_ = boost::shared_ptr<BusinessCenter>(new BusinessCenter(weatherStationCityNode));
             weatherStationCity_->setID(weatherStationCityIDRef_);
             IDManager::instance().setID(weatherStationCityIDRef_,weatherStationCity_);
          }
          else if(weatherStationCityNode->Attribute("href")) { weatherStationCityIDRef_ = weatherStationCityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { weatherStationCity_ = boost::shared_ptr<BusinessCenter>(new BusinessCenter(weatherStationCityNode));}
   }

   //weatherStationAirportNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* weatherStationAirportNode = xmlNode->FirstChildElement("weatherStationAirport");

   if(weatherStationAirportNode){weatherStationAirportIsNull_ = false;}
   else{weatherStationAirportIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- weatherStationAirportNode , address : " << weatherStationAirportNode << std::endl;
   #endif
   if(weatherStationAirportNode)
   {
      if(weatherStationAirportNode->Attribute("href") || weatherStationAirportNode->Attribute("id"))
      {
          if(weatherStationAirportNode->Attribute("id"))
          {
             weatherStationAirportIDRef_ = weatherStationAirportNode->Attribute("id");
             weatherStationAirport_ = boost::shared_ptr<WeatherStationAirport>(new WeatherStationAirport(weatherStationAirportNode));
             weatherStationAirport_->setID(weatherStationAirportIDRef_);
             IDManager::instance().setID(weatherStationAirportIDRef_,weatherStationAirport_);
          }
          else if(weatherStationAirportNode->Attribute("href")) { weatherStationAirportIDRef_ = weatherStationAirportNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { weatherStationAirport_ = boost::shared_ptr<WeatherStationAirport>(new WeatherStationAirport(weatherStationAirportNode));}
   }

   //weatherStationWBANNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* weatherStationWBANNode = xmlNode->FirstChildElement("weatherStationWBAN");

   if(weatherStationWBANNode){weatherStationWBANIsNull_ = false;}
   else{weatherStationWBANIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- weatherStationWBANNode , address : " << weatherStationWBANNode << std::endl;
   #endif
   if(weatherStationWBANNode)
   {
      if(weatherStationWBANNode->Attribute("href") || weatherStationWBANNode->Attribute("id"))
      {
          if(weatherStationWBANNode->Attribute("id"))
          {
             weatherStationWBANIDRef_ = weatherStationWBANNode->Attribute("id");
             weatherStationWBAN_ = boost::shared_ptr<WeatherStationWBAN>(new WeatherStationWBAN(weatherStationWBANNode));
             weatherStationWBAN_->setID(weatherStationWBANIDRef_);
             IDManager::instance().setID(weatherStationWBANIDRef_,weatherStationWBAN_);
          }
          else if(weatherStationWBANNode->Attribute("href")) { weatherStationWBANIDRef_ = weatherStationWBANNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { weatherStationWBAN_ = boost::shared_ptr<WeatherStationWBAN>(new WeatherStationWBAN(weatherStationWBANNode));}
   }

   //weatherStationWMONode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* weatherStationWMONode = xmlNode->FirstChildElement("weatherStationWMO");

   if(weatherStationWMONode){weatherStationWMOIsNull_ = false;}
   else{weatherStationWMOIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- weatherStationWMONode , address : " << weatherStationWMONode << std::endl;
   #endif
   if(weatherStationWMONode)
   {
      if(weatherStationWMONode->Attribute("href") || weatherStationWMONode->Attribute("id"))
      {
          if(weatherStationWMONode->Attribute("id"))
          {
             weatherStationWMOIDRef_ = weatherStationWMONode->Attribute("id");
             weatherStationWMO_ = boost::shared_ptr<WeatherStationWMO>(new WeatherStationWMO(weatherStationWMONode));
             weatherStationWMO_->setID(weatherStationWMOIDRef_);
             IDManager::instance().setID(weatherStationWMOIDRef_,weatherStationWMO_);
          }
          else if(weatherStationWMONode->Attribute("href")) { weatherStationWMOIDRef_ = weatherStationWMONode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { weatherStationWMO_ = boost::shared_ptr<WeatherStationWMO>(new WeatherStationWMO(weatherStationWMONode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<BusinessCenter> WeatherStation::getWeatherStationCity()
{
   if(!this->weatherStationCityIsNull_){
        if(weatherStationCityIDRef_ != ""){
             return boost::shared_static_cast<BusinessCenter>(IDManager::instance().getID(weatherStationCityIDRef_));
        }else{
             return this->weatherStationCity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<BusinessCenter>();
   }
}
boost::shared_ptr<WeatherStationAirport> WeatherStation::getWeatherStationAirport()
{
   if(!this->weatherStationAirportIsNull_){
        if(weatherStationAirportIDRef_ != ""){
             return boost::shared_static_cast<WeatherStationAirport>(IDManager::instance().getID(weatherStationAirportIDRef_));
        }else{
             return this->weatherStationAirport_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<WeatherStationAirport>();
   }
}
boost::shared_ptr<WeatherStationWBAN> WeatherStation::getWeatherStationWBAN()
{
   if(!this->weatherStationWBANIsNull_){
        if(weatherStationWBANIDRef_ != ""){
             return boost::shared_static_cast<WeatherStationWBAN>(IDManager::instance().getID(weatherStationWBANIDRef_));
        }else{
             return this->weatherStationWBAN_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<WeatherStationWBAN>();
   }
}
boost::shared_ptr<WeatherStationWMO> WeatherStation::getWeatherStationWMO()
{
   if(!this->weatherStationWMOIsNull_){
        if(weatherStationWMOIDRef_ != ""){
             return boost::shared_static_cast<WeatherStationWMO>(IDManager::instance().getID(weatherStationWMOIDRef_));
        }else{
             return this->weatherStationWMO_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<WeatherStationWMO>();
   }
}
}

